### PR TITLE
oci-iscsi-config usage refactor

### PR DESCRIPTION
**Pull request by ejannett**
refactor usage to have more user friendly experience and simplify the code
```
sudo oci-iscsi-config --help
usage: oci-iscsi-config-main.py [-h]
                                {sync,usage,show,create,attach,detach,destroy}
                                ...

Utility for listing or configuring iSCSI devices on an OCI instance.

positional arguments:
  {sync,usage,show,create,attach,detach,destroy}

optional arguments:
  -h, --help            show this help message and exit
```
* For backward compatibility , the oci-iscsi-config script translate old usage to the new one.
* new option 
  1.  --yes when destruction happen
  2. --attach-volume 

test done:
``
- use former command line to verify they still works
-- #/usr/bin/oci-iscsi-config --interactive
-- #/usr/bin/oci-iscsi-config —show
-- # oci-iscsi-config --detach iqn.2015-12.com.oracleiaas:f80249cc-fb9d-44b7-88b1-0cf9fa92466b
-- #oci-iscsi-config --attach ocid1.volume.oc1.uk-london-1.abwgiljrobmtlskmbvvkbivxyalceuhywxiynh5uamz6cigc74i6ytpjlf7a --show
```
-use new command line
```
-- #sudo oci-iscsi-config create --size 128 --volume-name manu-test-vol_00 --show
-- #sudo oci-iscsi-config create --size 128 --volume-name manu-test-vol_01 --volume-attach
-- #sudo oci-iscsi-config create --size 128 --volume-name manu-test-vol_01 --attach-volume
-- #sudo oci-iscsi-config destroy -s --ocids ocid1.volume.oc1.uk-london1.abwgiljr3btwvdpbilun5jibcsmgjmapt2lf6wzxvb3wxl37qhhlfxcdztdq 
-- #sudo oci-iscsi-config show
--#sudo oci-iscsi-config detach --iqns qn.2015-12.com.oracleiaas:48b253ee-21b1-4a5e-bebf-ddca9f269497,iqn.2015-12.com.oracleiaas:eef3e23c-ae86-440f-ace1-2c3980cbe19a
--#sudo oci-iscsi-config attach -I ocid1.volume.oc1.uk-london-1.abwgiljrsoxmdhjzxpkrwzrwpaup4pimqi4k6mrgyeivupitvl4vszlw3aeq,ocid1.volume.oc1.uk-london-1.abwgiljrtffarqr4jxgiqoqj57panbjv6a35jmgzrbuoqd4cuiza2em4525q
```

### CODE REVIEW

**Review by samiam69 — APPROVED** [review]

File: lib/oci_utils/impl/oci-iscsi-config-main.py [hunk 1]
```diff
@@ -24,13 +24,57 @@
...
+def attachable_iqn_list_validator(value):
+    """
+    validate than value passed is a list of iqn
+    """
+    _iqns =  [iqn.strip() for iqn in value.split(',')  if iqn]
+    for iqn in _iqns:
+        if not iqn.startswith("iqn.") and not iqn.startswith('ocid1.volume.oc'):
+            raise argparse.ArgumentTypeError('Invalid IQN %s' % iqn)
+    return _iqns
+def detachable_iqn_list_validator(value):
```
Comment (guidotijskens) [inline lib/oci_utils/impl/oci-iscsi-config-main.py:55]: 2 blank lines

File: lib/oci_utils/impl/oci-iscsi-config-main.py [hunk 2]
```diff
@@ -24,13 +24,57 @@
...
+    try:
+        _i_value = int(value)
+    except ValueError as e :
+        raise argparse.ArgumentTypeError("block volume size must be a int") from e
+
+    if _i_value < 50:
+        raise argparse.ArgumentTypeError("Volume size must be at least 50GBs")
+    return _i_value
+
+def attachable_iqn_list_validator(value):
```
Comment (guidotijskens) [inline lib/oci_utils/impl/oci-iscsi-config-main.py:46]: 1 extra blank line

File: lib/oci_utils/impl/oci-iscsi-config-main.py [hunk 3]
```diff
@@ -24,13 +24,57 @@
...
+    """
+    _iqns =  [iqn.strip() for iqn in value.split(',')  if iqn]
+    for iqn in _iqns:
+        if not iqn.startswith("iqn."):
+            raise argparse.ArgumentTypeError('Invalid IQN %s' % iqn)
+        if 'boot:uefi' in iqn:
+            raise argparse.ArgumentTypeError('Cannot detach boot volume IQN %s' % iqn)
+    return _iqns
+
+def volume_oci_list_validator(value):
```
Comment (guidotijskens) [inline lib/oci_utils/impl/oci-iscsi-config-main.py:67]: 1 extra blank line

File: lib/oci_utils/impl/oci-iscsi-config-main.py [hunk 4]
```diff
@@ -24,13 +24,57 @@
...
+    """
+    validate than value passed is a list of volume ocid
+    """
+    _ocids =  [ocid.strip() for ocid in value.split(',')  if ocid]
+    for ocid in _ocids:
+        if not ocid.startswith('ocid1.volume.oc'):
+            raise argparse.ArgumentTypeError('Invalid volume OCID %s' % ocid)
+    return _ocids
```
Comment (guidotijskens) [inline lib/oci_utils/impl/oci-iscsi-config-main.py:76]: blank line

File: lib/oci_utils/impl/oci-iscsi-config-main.py [hunk 5]
```diff
@@ -164,49 +207,56 @@ def ocid_refresh(wait=False, debug=False):
...
-                                              'iscsi'] + debug_opt,
-                                             stderr=subprocess.STDOUT)
-        _logger.debug(str(output))
+        output = subprocess.check_output(_cmd, stderr=subprocess.STDOUT)
+        if _logger.isEnabledFor(logging.DEBUG):
+            _logger.debug('ocid run output: %s' % str(output))
         return True
-    except subprocess.CalledProcessError:
+    except subprocess.CalledProcessError as e :
+        _logger.debug('launch of ocid failed : %s',str(e))
```
Comment (guidotijskens) [inline lib/oci_utils/impl/oci-iscsi-config-main.py:225]: shouldn't this be a _logger.error?

File: lib/oci_utils/impl/oci-iscsi-config-main.py [hunk 6]
```diff
@@ -218,79 +268,73 @@ def display_current_devices(oci_sess, session, disks):
...
-            oci_vol = None
-            for vol in oci_vols:
-                if vol.get_iqn() == iqn:
-                    oci_vol = vol
-                    break
+    try:
+        oci_vols = oci_sess.this_instance().all_volumes()
+    except Exception as e:
+        oci_vols = []
+        _logger.debug('Cannot get all volumes of this instance : %s' , str(e))
```
Comment (guidotijskens) [inline lib/oci_utils/impl/oci-iscsi-config-main.py:275]: _logger.error or _logger.warning???

**Review by guidotijskens — APPROVED** [review]
quite a job..

File: lib/oci_utils/impl/oci-iscsi-config-main.py (hunk 5 above)
Comment (ejannett) [reply lib/oci_utils/impl/oci-iscsi-config-main.py:225]: why not. I've followed logging already present . I'll change it

**Pull request merged**